indico/modules/users/controllers.py

```diff
@@ -33,32 +33,53 @@
 from indico.modules.auth.util import register_user
 from indico.modules.categories import Category
 from indico.modules.events import Event
+from indico.modules.events.util import serialize_event_for_ical
 from indico.modules.users import User, logger, user_management_settings
 from indico.modules.users.forms import (AdminAccountRegistrationForm, AdminsForm, AdminUserSettingsForm, MergeForm,
                                         SearchForm, UserDetailsForm, UserEmailsForm, UserPreferencesForm)
 from indico.modules.users.models.emails import UserEmail
 from indico.modules.users.operations import create_user
 from indico.modules.users.util import (get_linked_events, get_related_categories, get_suggested_categories, merge_users,
                                        search_users, serialize_user)
-from indico.modules.users.views import WPUser, WPUsersAdmin
-from indico.util.date_time import now_utc, timedelta_split
+from indico.modules.users.views import WPUser, WPUserDashboard, WPUsersAdmin
+from indico.util.date_time import now_utc
 from indico.util.event import truncate_path
 from indico.util.i18n import _
-from indico.util.marshmallow import validate_with_message
+from indico.util.marshmallow import HumanizedDate, validate_with_message
 from indico.util.signals import values_from_signal
 from indico.util.string import make_unique_token
 from indico.web.args import use_kwargs
 from indico.web.flask.templating import get_template_module
-from indico.web.flask.util import url_for
+from indico.web.flask.util import send_file, url_for
 from indico.web.forms.base import FormDefaults
+from indico.web.http_api.metadata import Serializer
 from indico.web.rh import RHProtected
-from indico.web.util import jsonify_data, jsonify_form, jsonify_template
+from indico.web.util import is_signed_url_valid, jsonify_data, jsonify_form, jsonify_template
 
 
 IDENTITY_ATTRIBUTES = {'first_name', 'last_name', 'email', 'affiliation', 'full_name'}
 UserEntry = namedtuple('UserEntry', IDENTITY_ATTRIBUTES | {'profile_url', 'user'})
 
 
+def get_events_in_categories(category_ids, user, limit=None):
+    """Get all the user-accessible events in a given set of categories."""
+    tz = session.tzinfo
+    today = now_utc(False).astimezone(tz).date()
+    query = (Event.query
+             .filter(~Event.is_deleted,
+                     Event.category_chain_overlaps(category_ids),
+                     Event.start_dt.astimezone(session.tzinfo) >= today)
+             .options(joinedload('category').load_only('id', 'title'),
+                      joinedload('series'),
+                      subqueryload('acl_entries'),
+                      load_only('id', 'category_id', 'start_dt', 'end_dt', 'title', 'access_key',
+                                'protection_mode', 'series_id', 'series_pos', 'series_count'))
+             .order_by(Event.start_dt, Event.id))
+    if limit:
+        query = query.limit(limit)
```

Comment:
I don't think this is correct. `get_n_matching` takes care of it, so you probably want to pass the limit to that instead of the hardcoded `10` (and probably use `10` as the default value for `limit`).